modules/analysis/src/service/mod.rs

```diff
@@ -38,7 +38,7 @@ use uuid::Uuid;
 
 #[derive(Clone)]
 pub struct AnalysisService {
-    graph: Arc<RwLock<GraphMap>>,
+    graph_cache: Arc<GraphMap>,
```

Comment:
I would wait till @ctron reviews ... previously he reworked things in this area and it was a bit tricky.